src/Sylius/Behat/Context/Ui/ShippingContext.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Ui;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Page\Shipping\ShippingMethodDetailsPageInterface;
+use Sylius\Behat\Page\UnexpectedPageException;
+use Sylius\Component\Core\Model\ShippingMethodInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class ShippingContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $shippingMethodRepository;
+
+    /**
+     * @var ShippingMethodDetailsPageInterface
+     */
+    private $shippingMethodDetailsPage;
+
+    /**
+     * @param RepositoryInterface $shippingMethodRepository
+     * @param SharedStorageInterface $sharedStorage
+     * @param ShippingMethodDetailsPageInterface $shippingMethodDetailsPage
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        RepositoryInterface $shippingMethodRepository,
+        ShippingMethodDetailsPageInterface $shippingMethodDetailsPage
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->shippingMethodRepository = $shippingMethodRepository;
+        $this->shippingMethodDetailsPage = $shippingMethodDetailsPage;
+    }
+
+    /**
+     * @When /^I try to delete ("[^"]+" shipping method)$/
+     */
+    public function iDeleteShippingMethod(ShippingMethodInterface $shippingMethod)
+    {
+        $this->shippingMethodDetailsPage->open(['id' => $shippingMethod->getId()]);
+        $this->shippingMethodDetailsPage->pressDelete();
+
+        $this->sharedStorage->set('shipping_method', $shippingMethod);
+    }
+
+    /**
+     * @Then I should be notified that it is in use
+     */
+    public function iShouldBeNotifiedItIsUsed()
+    {
+        $message = 'Cannot delete, the shipping method is in use.';
+
+        expect($this->shippingMethodDetailsPage->flashContainsMessage($message))->toBe(true);
+    }
+
+    /**
+     * @Then :it should be successfully removed
+     */
+    public function shippingMethodShouldBeRemoved(ShippingMethodInterface $shippingMethod)
+    {
+        expect($this->shippingMethodDetailsPage)
+            ->toThrow(UnexpectedPageException::class)
+            ->during('verify', [['id' => $shippingMethod->getId()]])
+        ;
+    }
+
+    /**
+     * @Then :it should not be removed
+     */
+    public function shippingMethodShouldNotBeRemoved(ShippingMethodInterface $shippingMethod)
+    {
+        expect($this->shippingMethodDetailsPage)
```

Comment:
`expect($this->shippingMethodDetailsPage)->isOpen(['id' => $shippingMethod->getId()]))->toBe(false)`.